docs/design_docs/011-historacle-pricing.md

```diff
@@ -0,0 +1,88 @@
+# Design Doc 011: Historacle Pricing
+
+## Status
+
+Draft
+
+## Abstract
+
+In order to support small-volume assets in a safe manner, we need to be able to calculate the weighted medians of our prices over the past X amount of time (30 days or so). This will allow the leverage module to make decisions around when to allow liquidations if an asset price is considered abnormal.
+
+## Context
+
+Right now, the volume of assets such as Juno, Umee, and Osmo are so low that it would be relatively easy to use the leverage protocol on UMEE to manipulate the prices of these assets and game the system.
+
+The attack goes:
+
+1. Account A deposits & collateralizes a large amount of USDC.
+2. Account A uses collateral to borrow a lot of the low-volume token (here, FOO).
+3. Account A sells the tokens to Account B, inflating the price of FOO.
+4. Account B collateralizes inflated FOO, and borrows USDC.
+5. Account B exits the market for a profit.
```

Comment:
We might use the classic scenario
```suggestion
1. Attacker deposits & collateralizes a large amount FOO, a token with low volume on exchanges.
2. Attacker spikes the price of FOO on the exchanges by buying a large amount.
3. Attacker borrows USDC using their FOO as collateral at its current (spiked) oracle price. 
4. The price of FOO returns to normal, and the value of the USDC exceeds the value of the FOO collateral.
5. Account B exits the market for a profit.
```

Reply:
There is also another scenario that might be included:

1. Attacker collateralizes USDC and borrows FOO
2. Attacker sells borrowed FOO on exchanges, dumping the price
3. Attacker borrows an even larger amount of FOO using the same collateral, due to the lower price
4. The price of FOO returns to normal, and the value of the FOO exceeds the value of the USDC collateral.
5. Attacker exits the market for a profit.

Reply:
I think it'd be good to have both. I'll add the 2nd scenario in the comments section.